pkg/rancher-components/src/components/Button/Button.vue

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+import { defineComponent, computed } from 'vue';
+
+type Props = {
+  primary: boolean;
+  secondary: boolean;
+  tertiary: boolean;
+};
+
+export default defineComponent({
+  props: {
+    primary:   Boolean,
+    secondary: Boolean,
+    tertiary:  Boolean,
+  },
+  setup(props) {
+    const buttonRoles = [
+      { role: 'primary', className: 'role-primary' },
```

Comment:
would it not be more convenient a dictionary?

Reply:
Maybe..

I chose an array of objects because I find order to be really important here. I want to assert that we maintain a hierarchy of styles from Primary => Secondary => Tertiary. 

Perhaps `Map`[^1] would better fit the bill here? Although, I find the ease of access using array methods to be quite convenient, but this is easily mitigated. 

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map